From 4d4ef68d88a652b5fd91ef08c22027bb65cc1009 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=A9my=20Bobbio?= <lunar@debian.org>
Date: Wed, 28 Aug 2013 18:22:44 +0200
Subject: [PATCH 3/3] Allow to preset dpkg-buildpackage timestamp

We now allow DEB_BUILD_TIMESTAMP to be externaly set before running
dpkg-buildpackage.

This enables to independently reproduce a previously built package by looking
at the Date field of its .changes file.
---
 scripts/dpkg-buildpackage.pl |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/dpkg-buildpackage.pl b/scripts/dpkg-buildpackage.pl
index 4173ad3..d9f45d3 100755
--- a/scripts/dpkg-buildpackage.pl
+++ b/scripts/dpkg-buildpackage.pl
@@ -124,7 +124,7 @@ my $binarytarget = 'binary';
 my $targetarch = my $targetgnusystem = '';
 my $call_target = '';
 my $call_target_as_root = 0;
-my $timestamp = time;
+my $timestamp = $ENV{DEB_BUILD_TIMESTAMP} || time;
 my (@checkbuilddep_opts, @changes_opts, @source_opts);
 
 use constant BUILD_DEFAULT    => 1;
-- 
1.7.10.4

